ab3d2_source/defs.i

```diff
@@ -211,6 +214,31 @@ LevelMusic			EQU		BackSFX+16*2
 EchoTable			EQU		LevelMusic+16*64
 LinkFileLen			EQU		EchoTable+60
 
+NUM_LEVELS			EQU	16
+NUM_BULLET_DEFS		EQU 20
+NUM_GUN_DEFS		EQU 10
+NUM_ALIEN_DEFS		EQU 20
+NUM_OBJECT_DEFS		EQU 30
+NUM_SFX				EQU 64
+
+	STRUCTURE GLFT,64
+		STRUCT GLFT_LevelNames_l,40*NUM_LEVELS
+		STRUCT GLFT_ObjGfxNames_l,(NUM_OBJECT_DEFS*64)
+		STRUCT GLFT_SFXFilenames_l,64*60
+		STRUCT GLFT_FloorFilename_l,64
+		STRUCT GLFT_TextureFilename_l,192
+		STRUCT GLFT_GunGFXFilename_l,64
+		STRUCT GLFT_StoryFilename_l,64
+		STRUCT GLFT_BulletDefs_l,(NUM_BULLET_DEFS*BulT_SizeOf_l)
+		STRUCT GLFT_BulletNames_l,(NUM_BULLET_DEFS*20)
+		STRUCT GLFT_GunNames_l,(NUM_GUN_DEFS*20)
+		STRUCT GLFT_ShootDefs_l,(NUM_GUN_DEFS*ShootT_SizeOf_l)
+		STRUCT GLFT_AlienNames_l,(NUM_ALIEN_DEFS*20)
+		STRUCT GLFT_AlienDefs_l,(NUM_ALIEN_DEFS*AlienT_SizeOf_l)
+		STRUCT GLFT_FrameData_l,7680 ; todo - figure out how this is derived
+		STRUCT GLFT_ObjectNames_l,(NUM_OBJECT_DEFS*20)
+		STRUCT GLFT_ObjectDefs,(NUM_OBJECT_DEFS*ObjT_SizeOf_l)
+
```

Comment:
This makes the structure of the link file so much more obvious!

Reply:
Yeah, it begins to demystify it a bit. One thing I wanted a second opinion about here is whether or not just making these offsets as _l makes sense or not. They are almost never used in a context where they'd be accessed as their implied "type", so even though it's obvious some of these are vectors of words, implying `_vw`, they're never typically accessed in a `my_array_vw(a0, d0.w*2)` addressing mode context. Instead it's usually that the offset is added to the address register first (or lea) as an immediate value. As such the size is down to the operation. Maybe it's better not to have a suffix for these.

Reply:
TBH, sub-structs probably don't need a suffix as they are typically used as an offset (most of the time less than a byte) only. We could use _s maybe to indicate sub-structs?
We might also get away without any type suffixes for regular struct members as they get 'typed' as part of the struct definition, but I'd argue that for consistency purposes with the rest of the code we should leave them.

Reply:
Based on your existing approval then, I'm going to merge it. We can always change these in a subsequent PR if we feel they are unweildy.